File: src/main/java/com/dongtronic/diabot/data/NightscoutDTO.java
package com.dongtronic.diabot.data;

import java.time.ZonedDateTime;

public class NightscoutDTO {
  private ConversionDTO glucose;
  private ConversionDTO delta;
  private ZonedDateTime dateTime;
  private boolean deltaIsNegative;
  private int low;
  private int bottom;
  private int top;
  private int high;

  public NightscoutDTO(ConversionDTO glucose, ConversionDTO delta, boolean deltaIsNegative, ZonedDateTime dateTime) {
    this.glucose = glucose;
    this.delta = delta;
    this.deltaIsNegative = deltaIsNegative;
    this.dateTime = dateTime;
  }

  public NightscoutDTO() {}


  //region properties
  public ConversionDTO getGlucose() {
    return glucose;
  }

  public void setGlucose(ConversionDTO glucose) {
    this.glucose = glucose;
  }

  public ConversionDTO getDelta() {
    return delta;
  }

  public boolean getDeltaIsNegative() {
    return deltaIsNegative;
  }

  public void setDelta(ConversionDTO delta) {
    this.delta = delta;
  }

  public void setDeltaIsNegative(boolean deltaIsNegative) {
    this.deltaIsNegative = deltaIsNegative;
  }

  public ZonedDateTime getDateTime() {
    return dateTime;
  }

  public void setDateTime(ZonedDateTime dateTime) {
    this.dateTime = dateTime;
  }

  public int getLow() {
    return low;
  }

  public void setLow(int low) {
    this.low = low;
  }

  public int getBottom() {
    return bottom;
  }

  public void setBottom(int bottom) {
    this.bottom = bottom;
  }

  public int getTop() {
    return top;
  }

  public void setTop(int top) {
    this.top = top;
  }

  public int getHigh() {
    return high;
  }

  public void setHigh(int high) {
    this.high = high;
  }
  //endregion
}

File: src/main/java/com/dongtronic/diabot/data/NightscoutDTO.kt
package com.dongtronic.diabot.data

import java.time.ZonedDateTime

class NightscoutDTO {
    var glucose: ConversionDTO? = null
    var delta: ConversionDTO? = null
    var dateTime: ZonedDateTime? = null
    var deltaIsNegative: Boolean = false
    var low: Int = 0
    var bottom: Int = 0
    var top: Int = 0
    var high: Int = 0
    var trend: Int = 0
    var iob: Float = 0.0F
    var cob: Int = 0
    var units: String = ""
    var title: String = "Nightscout"
    set (value) {
        if (value != null) {
            field = value
        }
    }

}
